src/scripts/core/models/grid.ts

```diff
@@ -65,10 +65,57 @@ export class Grid extends Container {
 
     }
 
+    private matchCheck(inputCoords: number[]): void {
+
+        const originX = inputCoords[0];
+        const originY = inputCoords[1];
+        const origin = this.getToken(originX, originY);
+        let xMatches = []; xMatches.push(origin);
+        let yMatches = []; yMatches.push(origin);
+
+        //check token's right
+        for(let i = originX+1; i <= this.gridSize-1; i++) {
+            if(!this.isMatch(origin, this.getToken(i, originY))) {break;}
+            xMatches.push(this.getToken(i, originY))
+        }
+
+        //token's left
+        for(let i = originX-1; i >= 0; i--) {
+            if(!this.isMatch(origin, this.getToken(i, originY))) {break;}
+            xMatches.push(this.getToken(i, originY))
+        }
+
+        //beneath token
+        for(let i = originY+1; i <= this.gridSize-1; i++) {
+            if(!this.isMatch(origin, this.getToken(originX, i))) {break;}
+            yMatches.push(this.getToken(originX, i))
+        }
+
+        //above token
+        for(let i = originY-1; i >= 0; i--) {
+            if(!this.isMatch(origin, this.getToken(originX, i))) {break;}
+            yMatches.push(this.getToken(originX, i))
+        }
+
+        console.log(xMatches, yMatches);
```

Comment:
If you still need logs like this to help with development feel free to leave them, otherwise please try and clean them up as you go